lib/common/graph.js

```diff
@@ -219,8 +219,13 @@ class ConcertoGraphVisitor extends DiagramVisitor {
             if(argument && argument.name) {
                 const fqn = modelFile.getFullyQualifiedTypeName(argument.name);
                 if (fqn) {
-                    parameters.graph.addVertex(fqn);
-                    (parent === modelFile) ? parameters.graph.addEdge(parent.getNamespace(), fqn) : parameters.graph.addEdge(parent.getFullyQualifiedName(), fqn);
+                    const type = modelFile.getModelManager().getType(fqn);
+                    const parentFqn = parent === modelFile ? parent.getNamespace() : parent.getFullyQualifiedName();
+                    if (parent instanceof Property) {
+                        parameters.graph.addEdge(parameters.stack.slice(-1), type.getFullyQualifiedName());
+                    } else {
```

Comment:
I'm struggling to understand this condition, can you add a comment to explain, please?
If this is an alternative way of calculating the parent node, perhaps you could combine with the previous line 223?

Is this equivalent to checking if we're visiting a Decorator node?

Reply:
> If this is an alternative way of calculating the parent node

Yes, this is an alternative approach to calculating the parent node. However, we cannot combine the method because 'properties' require handling that differs from Declarations and ModelFiles.

In the condition `if (parent instanceof Property) {} else {}`, the goal is to establish an edge that represents the relationship of a decorator, particularly in these cases:
- When the decorated node is a ScalarDeclaration or MapDeclaration
- When the decorated node is a Field or EnumValueDeclaration

Currently, these relationships, especially those involving fields, are not accurately represented. The link between the parent declaration and the field is missing, which leads to errors when attempting to findConnectedGraph, as these decorators are not retained in the graph structure.

This approach also addresses the previously identified issue, where a decorator could not resolve back to its source declaration. The earlier reliance on the graph pattern to identify the source node left some connections unresolved, but this adjustment ensures the decorator can consistently trace back to the correct source declaration.